src/QsCompiler/LanguageServer/LanguageServer.cs

```diff
@@ -185,6 +185,12 @@ internal async Task CheckDotNetSdkVersionAsync()
         /// </summary>
         internal void OnInternalError(Exception ex)
         {
+            var telemetryProps = new Dictionary<string, string?>();
+            telemetryProps["exceptionType"] = ex.GetType().ToString();
+            telemetryProps["message"] = ex.Message;
```

Comment:
Nit
```suggestion
            var telemetryProps = new Dictionary<string, string?>()
            {
                { "exceptionType" } = ex.GetType().FullName;
            }
```

Reply:
It would be nice to capture the Exception message for easier investigations, but it can contain customer data / PII and therefore we should avoid collecting it.

Reply:
That's a good point, thanks. It is a bit unfortunate since the stack trace is what we would need to fix the issue, but I don't see a good way to get more information. I'll remove it.

Reply:
I wrote a [(private static) method](https://github.com/microsoft/qsharp-compiler/blob/8da3079da3c5387f12e0dfd059d4cb4ab6cf0ca2/src/Telemetry/Library/TypeConversionHelper.cs#L197) in a Telemetry Library helper class to sanitize the StackTrace of an exception.
If you want to take a dependency on the Telemetry Library, you could just make that method public and use it.
Another option would be to copy that code, but I would prefer to have it in a single place to make it easier to fix bugs in the future.